### PR TITLE
Add destroy, disable, enable ops for azure server groups

### PR DESCRIPTION
**Pull request by milismsft**
@JargoonPard PTAL

### CODE REVIEW

File: app/scripts/modules/azure/serverGroup/details/serverGroupDetails.html [hunk 1]
```diff
@@ -27,32 +27,32 @@ <h3 select-on-dbl-click>
...
+        </button>
           <ul class="dropdown-menu" uib-dropdown-menu role="menu">
+		  <div ng-if="false">
             <li><a href ng-click="ctrl.resizeServerGroup()">Resize</a></li>
-            <li><a href ng-if="!serverGroup.isDisabled" ng-click="ctrl.disableServerGroup()">Disable</a></li>
-            <li><a href ng-if="serverGroup.isDisabled" ng-click="ctrl.enableServerGroup()">Enable</a></li>
-            <li><a href ng-click="ctrl.destroyServerGroup()">Destroy</a></li>
+		  </div>
+          <li><a href ng-click="ctrl.disableServerGroup()">Disable</a></li>
+          <li><a href ng-click="ctrl.enableServerGroup()">Enable</a></li>
```
Comment (scotmoor) [inline app/scripts/modules/azure/serverGroup/details/serverGroupDetails.html]: Is there a reason we are not checking the disable/enabled state anymore?

File: app/scripts/modules/azure/serverGroup/details/serverGroupDetails.html [hunk 2]
```diff
@@ -27,32 +27,32 @@ <h3 select-on-dbl-click>
...
+          <li><a href ng-click="ctrl.destroyServerGroup()">Destroy</a></li>
+		  <div ng-if="false">
```
Comment (scotmoor) [inline app/scripts/modules/azure/serverGroup/details/serverGroupDetails.html]: What does just setting this to 'false' do?

File: app/scripts/modules/azure/serverGroup/details/serverGroupDetails.azure.controller.js [hunk 3]
```diff
@@ -110,12 +110,12 @@ module.exports = angular.module('spinnaker.azure.serverGroup.details.controller'
         application: app,
         title: 'Destroying ' + serverGroup.name,
         forceRefreshMessage: 'Refreshing application...',
-        forceRefreshEnabled: true,
-        katoPhaseToMonitor: 'DESTROY_ASG'
+        forceRefreshEnabled: true
+        // katoPhaseToMonitor: 'DESTROY_ASG'
```
Comment (scotmoor) [inline app/scripts/modules/azure/serverGroup/details/serverGroupDetails.azure.controller.js]: Should this and the other commented out lines just be removed?

File: app/scripts/modules/azure/serverGroup/details/serverGroupDetails.azure.controller.js [hunk 4]
```diff
@@ -10,12 +10,12 @@ module.exports = angular.module('spinnaker.azure.serverGroup.details.controller'
   require('../../../core/serverGroup/serverGroup.read.service.js'),
   require('../../../core/utils/selectOnDblClick.directive.js'),
   require('../../../core/confirmationModal/confirmationModal.service.js'),
-  require('../../../core/serverGroup/serverGroup.write.service.js'),
+  require('../../../azure/serverGroup/serverGroup.write.service.js'),
```
Comment (JargoonPard) [inline app/scripts/modules/azure/serverGroup/details/serverGroupDetails.azure.controller.js]: Simplify the path. You don't need to go up the tree quite that far to reference that file.

**Comment by JargoonPard** [comment]
Until we get the backend capabilities built for enable and disable those actions should not be added into this commit. Remove the functions and drop down entries that are not part of the active work.

File: app/scripts/modules/azure/serverGroup/serverGroup.write.service.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    require('../../core/task/taskExecutor.js'),
+    require('./serverGroup.transformer.js'),
+  ])
+  .factory('azureServerGroupWriter', function (taskExecutor, serverGroupTransformer) {
+
+    function destroyServerGroup(serverGroup, application, params = {}) {
+      params.serverGroupName = serverGroup.name;
+      params.asgName = serverGroup.name;
+      params.type = 'destroyServerGroup';
+      params.regions = [serverGroup.region];  // Needed for AWS
```
Comment (JargoonPard) [inline app/scripts/modules/azure/serverGroup/serverGroup.write.service.js]: We shouldn't have field that are only for AWS

File: app/scripts/modules/azure/serverGroup/serverGroup.write.service.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+      params.region = serverGroup.region;     // Needed for Titan, GCE
```
Comment (JargoonPard) [inline app/scripts/modules/azure/serverGroup/serverGroup.write.service.js]: We shouldn't have fields that are only for Titan and GCE

File: app/scripts/modules/azure/serverGroup/serverGroup.write.service.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+      params.cloudProvider = serverGroup.type;
+
+      return taskExecutor.executeTask({
+        job: [params],
+        application: application,
+        description: 'Destroy Server Group: ' + serverGroup.name
+      });
+    }
+
+    function disableServerGroup(serverGroup, applicationName, params = {}) {
```
Comment (JargoonPard) [inline app/scripts/modules/azure/serverGroup/serverGroup.write.service.js]: Remove this function

File: app/scripts/modules/azure/serverGroup/serverGroup.write.service.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+      params.cloudProvider = serverGroup.type;
+
+      return taskExecutor.executeTask({
+        job: [params],
+        application: applicationName,
+        description: 'Disable Server Group: ' + serverGroup.name
+      });
+    }
+
+    function enableServerGroup(serverGroup, application, params = {}) {
```
Comment (JargoonPard) [inline app/scripts/modules/azure/serverGroup/serverGroup.write.service.js]: Remove this function

File: app/scripts/modules/azure/serverGroup/serverGroup.write.service.js [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+      params.cloudProvider = serverGroup.type;
+
+      return taskExecutor.executeTask({
+        job: [params],
+        application: application,
+        description: 'Enable Server Group: ' + serverGroup.name
+      });
+    }
+
+    function rollbackServerGroup(serverGroup, application, params = {}) {
```
Comment (JargoonPard) [inline app/scripts/modules/azure/serverGroup/serverGroup.write.service.js]: Remove this function

File: app/scripts/modules/azure/serverGroup/serverGroup.write.service.js [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+      params.cloudProvider = serverGroup.type;
+
+      return taskExecutor.executeTask({
+        job: [params],
+        application: application,
+        description: 'Rollback Server Group: ' + serverGroup.name
+      });
+    }
+
+    function resizeServerGroup(serverGroup, application, params = {}) {
```
Comment (JargoonPard) [inline app/scripts/modules/azure/serverGroup/serverGroup.write.service.js]: Remove this function

File: app/scripts/modules/azure/serverGroup/serverGroup.write.service.js [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+      params.credentials = serverGroup.account;
+      params.cloudProvider = serverGroup.type;
+      return taskExecutor.executeTask({
+        job: [params],
+        application: application,
+        description: 'Resize Server Group: ' + serverGroup.name + ' to ' + params.capacity.min + '/' + params.capacity.desired + '/' + params.capacity.max
+      });
+    }
+
+    function cloneServerGroup(command, application) {
```
Comment (JargoonPard) [inline app/scripts/modules/azure/serverGroup/serverGroup.write.service.js]: Remove this function

File: app/scripts/modules/azure/serverGroup/serverGroup.write.service.js [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+let angular = require('angular');
+
+module.exports = angular
+  .module('spinnaker.azure.AzureServerGroup.write.service', [
+    require('../../core/task/taskExecutor.js'),
+    require('./serverGroup.transformer.js'),
+  ])
+  .factory('azureServerGroupWriter', function (taskExecutor, serverGroupTransformer) {
+
+    function destroyServerGroup(serverGroup, application, params = {}) {
```
Comment (JargoonPard) [inline app/scripts/modules/azure/serverGroup/serverGroup.write.service.js]: How is this function different from the one in core? What did you change?

**Comment by JargoonPard** [comment]
I didn't see any changes that you made to the serverGroup.write.service.js file from the one that's in core.   Can you point them out please?

File: app/scripts/modules/azure/serverGroup/details/serverGroupDetails.html (hunk 1 above)
Comment (milismsft) [inline app/scripts/modules/azure/serverGroup/details/serverGroupDetails.html]: enable/disable check correspond to a property of the serverGroup (isDisabled) which needs to be properly implemented in the clouddriver and should be part of the AppGateway implementation discussion.

File: app/scripts/modules/azure/serverGroup/details/serverGroupDetails.html (hunk 2 above)
Comment (milismsft) [inline app/scripts/modules/azure/serverGroup/details/serverGroupDetails.html]: It disables the buttons selection; those actions will have to be implemented as some point.

File: app/scripts/modules/azure/serverGroup/details/serverGroupDetails.azure.controller.js (hunk 4 above)
Comment (milismsft) [inline app/scripts/modules/azure/serverGroup/details/serverGroupDetails.azure.controller.js]: will do

File: app/scripts/modules/azure/serverGroup/serverGroup.write.service.js (hunk 5 above)
Comment (milismsft) [inline app/scripts/modules/azure/serverGroup/serverGroup.write.service.js]: it's not just for AWS; we also expect the region as a list. But I will remove the comment.

File: app/scripts/modules/azure/serverGroup/serverGroup.write.service.js (hunk 6 above)
Comment (milismsft) [inline app/scripts/modules/azure/serverGroup/serverGroup.write.service.js]: I will remove that

File: app/scripts/modules/azure/serverGroup/serverGroup.write.service.js (hunk 7 above)
Comment (milismsft) [inline app/scripts/modules/azure/serverGroup/serverGroup.write.service.js]: This function can still be called into (see the HTML related change).

File: app/scripts/modules/azure/serverGroup/serverGroup.write.service.js (hunk 8 above)
Comment (milismsft) [inline app/scripts/modules/azure/serverGroup/serverGroup.write.service.js]: same as above.

File: app/scripts/modules/azure/serverGroup/serverGroup.write.service.js (hunk 9 above)
Comment (milismsft) [inline app/scripts/modules/azure/serverGroup/serverGroup.write.service.js]: We will need this soon (there's a card to add cloning support). You sure you want it removed?

File: app/scripts/modules/azure/serverGroup/serverGroup.write.service.js (hunk 10 above)
Comment (milismsft) [inline app/scripts/modules/azure/serverGroup/serverGroup.write.service.js]: Same here.

File: app/scripts/modules/azure/serverGroup/serverGroup.write.service.js (hunk 11 above)
Comment (milismsft) [inline app/scripts/modules/azure/serverGroup/serverGroup.write.service.js]: and here

File: app/scripts/modules/azure/serverGroup/serverGroup.write.service.js (hunk 12 above)
Comment (milismsft) [inline app/scripts/modules/azure/serverGroup/serverGroup.write.service.js]: the "core" implementation also sets "zones" param; I also tried removing the asgName but unfortunately there's code in orca core functionality that expects that property today and I did not want to go too deep and change that.

**Comment by JargoonPard** [comment]
LGTM

File: app/scripts/modules/azure/serverGroup/details/serverGroupDetails.azure.controller.js (hunk 3 above)
Comment (scotmoor) [inline app/scripts/modules/azure/serverGroup/details/serverGroupDetails.azure.controller.js]: Actually, do we fully understand what commenting this out does? Given the name, It seems to imply that the UI has some logic to monitor a task and update accordingly. By commenting this out, are we unknowingly disabling functionality we want?

Comment (milismsft) [inline app/scripts/modules/azure/serverGroup/details/serverGroupDetails.azure.controller.js]: The phase to monitor for us is called 'DESTROY_SERVER_GROUP'; this seems to be some legacy thing left around that works only for AWS. I tested the code through the deck without it number of times, it works as expected.
screenshot from 2016-04-13 10 34 31

**Pull request closed without merging**